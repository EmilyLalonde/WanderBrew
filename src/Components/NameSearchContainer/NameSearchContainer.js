import React, { Component } from 'react'
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { getNameResults } from '../../actions'
import './NameSearchContainer.css'
import { getBreweriesByName } from "../../apiCalls/apiCalls";
import NameContainer from '../NameContainer/NameContainer';

class NameSearchContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      error: ''
    }
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmitName = async () => {
    this.setState({ name: "" });
    const { name } = this.state;
    const { getNameResults } = this.props
    getBreweriesByName(name)
    .then(data => getNameResults(data))
  }
    // this.setState({ stateSearch: [] });
    // this.setState({ name: "" });
    // try {
    //   const nameSearch = await getBreweriesByName(this.state.name)
    //   this.setState({ nameSearch })
    // } catch ({ message }) {
    //   this.setState({ error: message });
    // }
  

  render() {
    console.log(this.props)
    return (
        <div>
        <input
          type="text"
          placeholder="Search for a name..."
          name="name"
          className="search"
          value={this.state.name}
          onChange={this.handleChange}
        />
        <button className="search-button" onClick={this.handleSubmitName}>
          Submit
        </button>
        <NameContainer />
      </div>
    )
  }
}

const mapStateToProps = state => ({
  nameResults:state.nameResults
})

const mapDispatchToProps = dispatch => (
  bindActionCreators({ getNameResults }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(NameSearchContainer)