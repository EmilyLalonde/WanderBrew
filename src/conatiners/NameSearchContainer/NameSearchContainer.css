.name-input-container {
  border: 2px solid black;
  margin-left: 1%;
  margin-top: 5%;
  height: 46px;
  width: 45%
}