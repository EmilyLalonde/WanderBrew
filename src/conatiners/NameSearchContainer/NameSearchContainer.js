import React, { Component } from 'react'
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { getNameResults } from '../../actions'
import './NameSearchContainer.css'
import { getBreweriesByName } from "../../apiCalls/apiCalls";
import NameContainer from '../NameContainer/NameContainer';

class NameSearchContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      error: ''
    }
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmitName = async () => {
    this.setState({ name: "" });
    const { name } = this.state;
    const { getNameResults } = this.props
    getBreweriesByName(name)
    .then(data => getNameResults(data))
  }

  render() {
    console.log(this.props)
    return (
      <main className="name-main">
      <div className="name-input-container">
        <input
          type="text"
          placeholder="Search for a name..."
          name="name"
          className="search"
          value={this.state.name}
          onChange={this.handleChange}
        />
        <button className="search-button" onClick={this.handleSubmitName}>
          Submit
        </button>
      </div>
      <div className="name-search-container">
        <NameContainer />
        </div>
      </main>
    )
  }
}

const mapStateToProps = state => ({
  nameResults:state.nameResults
})

const mapDispatchToProps = dispatch => (
  bindActionCreators({ getNameResults }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(NameSearchContainer)