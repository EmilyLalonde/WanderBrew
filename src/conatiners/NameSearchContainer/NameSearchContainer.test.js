import React from "react";
import { shallow } from "enzyme";
import  { NameSearchContainer, mapStateToProps }  from './NameSearchContainer';

describe("NameSearchContainer", () => {
  let wrapper;
  let mockNameResults;

  beforeEach(() => {
    mockNameResults = [
      {
        id: 1516,
        name: "Ratio Beerworks",
        street: "2920 Larimer St",
        city: "Denver",
        state: "Colorado",
        phone: "3012669264"
      }
    ];
    wrapper = shallow(<NameSearchContainer props={mockNameResults}/>);
  });
  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should update state when handleChange is called", () => {
    wrapper = shallow(<NameSearchContainer />);
    const mockEvent = { target: { name: "state", value: "Colorado" } };
    const expected = "";

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state("name")).toEqual(expected);
  });

  it("should run handleChange when the inputs detect a change", () => {
    const mockEvent = { target: { name: "state", value: "Colorado" } };

    wrapper.instance().handleChange = jest.fn();
    wrapper.find('[name="name"]').simulate("change", mockEvent);

    expect(wrapper.instance().handleChange).toHaveBeenCalledTimes(0);
  });

  it("should run handleSubmitState when the button is clicked", () => {
    const mockEvent = { preventDefault: jest.fn() };

    wrapper.instance().handleSubmitState = jest.fn();
    wrapper.find("button").simulate("click", mockEvent);

    expect(wrapper.instance().handleSubmitState).toHaveBeenCalledTimes(0);
  });
});
