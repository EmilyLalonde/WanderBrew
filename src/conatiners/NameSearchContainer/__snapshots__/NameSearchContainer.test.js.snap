// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameSearchContainer should match the snapshot 1`] = `
<main
  className="name-main"
>
  <div
    className="name-input-container"
  >
    <input
      className="search"
      name="name"
      onChange={[Function]}
      placeholder="Search for a name..."
      type="text"
      value=""
    />
    <button
      className="search-button"
      disabled={true}
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="name-search-container"
  >
    <Connect(NameContainer) />
    <h4>
      ^ Search for Breweries by Name ^
    </h4>
  </div>
</main>
`;
